airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -140,24 +150,72 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
                 "gcp_conn_id": self.gcp_conn_id,
                 "impersonation_chain": self.impersonation_chain,
                 "polling_interval_seconds": self.polling_interval_seconds,
+                "delete_on_error": self.delete_on_error,
             },
         )
 
     async def run(self) -> AsyncIterator[TriggerEvent]:
-        while True:
-            cluster = await self.get_async_hook().get_cluster(
-                project_id=self.project_id, region=self.region, cluster_name=self.cluster_name
+        """Run the trigger."""
+        try:
+            while True:
+                cluster = await self.fetch_cluster()
+                state = cluster.status.state
+                if state == ClusterStatus.State.ERROR:
+                    await self.delete_when_error_occurred(cluster)
+                    yield TriggerEvent(
+                        {
+                            "cluster_name": self.cluster_name,
+                            "cluster_state": state.DELETING,
```

Comment:
Can we re-check here that `state.DELETING` is the right usage because `state=ClusterStatus.State.ERROR` after entering the if block here?

Reply:
Thats correct. It should be Error. Changed it

Reply:
No I meant it may not be `state.DELETING` or `state.ERROR` since `state=ClusterStatus.State.ERROR` after entering the if block. `state.ERROR` would mean `ClusterStatus.State.ERROR.ERROR`

Reply:
I have changed it to `ClusterStatus.State.DELETING`